### ISSUE TITLE
Migration Error: "t.getStructure is not a function"

### ISSUE THREAD

**Issue by Lavaei** [ngneat/until-destroy#117]
I've tested the migration tool on serveral projects and all of them done succesfully. But one of them throw an exception:

```
/*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:14470
            tags: this.getTags().map(t => t.getStructure()),
                                            ^

TypeError: t.getStructure is not a function
    at /*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:14470:45
    at Array.map (<anonymous>)
    at JSDoc.getStructure (/*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:14470:34)
    at /*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:8688:59
    at Array.map (<anonymous>)
    at MethodDeclaration.getStructure (/*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:8688:40)
    at callBaseGetStructure (/*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:534:51)
    at MethodDeclaration.getStructure (/*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:8470:20)
    at callBaseGetStructure (/*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:534:51)
    at MethodDeclaration.getStructure (/*****/*****/*****/*****/*****/*****/node_modules/ts-morph/dist/ts-morph.js:2771:20)
```

Thanks for your help.

**Comment by Lavaei** [comment]
I think is should add the only difference between this project and previous projects is that this project is an angular library and I give the path like this:

```
npx @ngneat/until-destroy --base ./projects/my-project/
```

**Comment by Lavaei** [comment]
I've found the problem and solve it using a dirty solution.
The problem is in the `ts-morph.js`:

```
// In the JSDoc class
    getStructure() {

        return callBaseGetStructure(JSDocBase.prototype, this, {
            kind: exports.StructureKind.JSDoc,
            description: this.getDescription(),
            tags: this.getTags().map(t => t.getStructure()),
        });
    }
```
In my case, `getTags()` return an array with just one element. that element is of type `Node`. As you can see in the `js-morph.js`, `Node` does not have `getStructure()` method, I get an exception here.
I've solved this problem by changing this method:
```
    getStructure() {

        return callBaseGetStructure(JSDocBase.prototype, this, {
            kind: exports.StructureKind.JSDoc,
            description: this.getDescription(),
            tags: this.getTags().filter(t => t.getStructure).map(t => t.getStructure()),
        });
    }
```
Now it works for me. But I know it is not a standard way to do it.
I don't no this problem is related to `until-destroy` package or `ts-morph`, so i keep this issue open.

**Comment by NetanelBasal** [comment]
@FFKL

**Comment by FFKL** [comment]
@Lavaei @NetanelBasal Ok, I will check in next week

**Comment by FFKL** [comment]
@Lavaei Can you provide a minimal reproducible example? I haven't enough info. Where the error occurred in `run.ts`?

**Comment by Lavaei** [comment]
I got the error a few seconds after running the migration. The error only occurred for two or three files. when I exclude them from the project, migration works correctly. I can't find the difference between those files and the others.
Here is a part of one of those files:
```
  constructor(protected _store: Store<RootState>)
  {
    super();
  }

  protected _getColorsFromSettings()
  {
    this._store.pipe(
      /**
       * Get settings
       */
      select(settingsSelector),

      /**
       * Unsubscribe on destroy
       */
      untilDestroyed(this)
    )
  }
```

I hope it helps.

**Comment by FFKL** [comment]
@Lavaei Thanks for this case, but I still can't catch the error;) It looks like `ts-morph` [bug](https://github.com/dsherret/ts-morph/issues/844)

**Comment by FFKL** [comment]
@NetanelBasal You can close this PR. It's already fixed in [#113](https://github.com/ngneat/until-destroy/issues/113). `classDeclaration.getStructure()` which triggers this error was removed.
